psycopg/psycopg/cursor.py

```diff
@@ -218,20 +224,28 @@ def _executemany_gen(
                 pgq.dump(params)
 
             results = yield from self._maybe_prepare_gen(pgq, prepare=True)
-            self._check_results(results)
-            if returning and results[0].status == ExecStatus.TUPLES_OK:
-                self._results.extend(results)
 
-            for res in results:
-                nrows += res.command_tuples or 0
+            if self._conn._pipeline:
+                yield from self._conn._pipeline._communicate_gen()
+            else:
+                assert results is not None
+                self._check_results(results)
+                if returning and results[0].status == ExecStatus.TUPLES_OK:
+                    self._results.extend(results)
```

Comment:
The management of results and rowcount in the pipeline block is broken with this change. I have fixed most of the most obvious problems in the [executemany-pipeline](https://github.com/psycopg/psycopg/tree/executemany-pipeline) branch, but the problem persists if executemany() is executed under a pipeline().

Maybe we could add a `self._fetch_pipeline()` call to nextset/rowcount? mmm, nah, it would worn in async mode... Anyway I wouldn't be desperate if it turns out that rowcount is unreliable in this case.

Reply:
I'm working on fixing `rowcount` with executemany() and pipeline; will probably send a PR on top of #260 today.